Fix the Naptr records serialization issue

WhiteHorseV opened the pull request:
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

WhiteHorseV committed:
Fix the Naptr records serialization issue

azure-sdk commented:
**API change check**

API changes are not detected in this pull request.

ArthurMa1978 commented on sdk/dns/Azure.ResourceManager.Dns/src/Customization/Models/DnsNaptrData.Serialization.cs:
```diff
@@ -82,7 +82,7 @@ void IJsonModel<DnsNaptrRecordData>.Write(Utf8JsonWriter writer, ModelReaderWrit
             if (Optional.IsCollectionDefined(DnsNaptrRecords))
             {
-                writer.WritePropertyName("NaptrRecords"u8);
+                writer.WritePropertyName("NAPTRRecords"u8);
```
Please add test for this change

WhiteHorseV replied:
Thanks @ArthurMa1978 Added the tests.
I was not able to add the tests as we needed some server side changes deployed.

WhiteHorseV committed:
Update DnsRecordDataTest.cs

WhiteHorseV committed:
Update RecordTest.cs

WhiteHorseV committed:
Remove trailing whitespaces

WhiteHorseV committed:
Remove trailing whitespace

archerzz reviewed (dismissed):
Need to update test recording. See the error logs: https://dev.azure.com/azure-sdk/public/_build/results?buildId=4265737&view=logs&j=19f517c9-ad26-5827-ca69-20f64049c15d&t=64a812af-f087-5e7b-c041-871b1ece27d6&l=1495

WhiteHorseV committed:
Merge branch 'Azure:main' into feature/naptrSerFixes

WhiteHorseV committed:
Update assets.json

WhiteHorseV commented:
> Need to update test recording. See the error logs: https://dev.azure.com/azure-sdk/public/_build/results?buildId=4265737&view=logs&j=19f517c9-ad26-5827-ca69-20f64049c15d&t=64a812af-f087-5e7b-c041-871b1ece27d6&l=1495

Updated the assets repo and updated the assets.json file.
